bindinfo/cache.go

```diff
@@ -191,3 +207,79 @@ func (b cache) appendNode(newBindRecord *bindRecord, sparser *parser.Parser) err
 	b[hash] = append(b[hash], newNode)
 	return nil
 }
+
+// AddGlobalBind implements GlobalBindAccessor.AddGlobalBind interface.
+func (h *Handle) AddGlobalBind(originSQL, bindSQL, defaultDB, charset, collation string) error {
+	h.lock.Lock()
+	defer h.lock.Unlock()
+
+	ctx := context.TODO()
+	exec, _ := h.ctx.(sqlexec.SQLExecutor)
+	_, err := exec.Execute(ctx, "BEGIN")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			_, err = exec.Execute(ctx, "COMMIT")
+		} else {
+			_, rbErr := exec.Execute(ctx, "ROLLBACK")
+			terror.Log(rbErr)
+		}
+	}()
+
+	sql := fmt.Sprintf("SELECT _tidb_rowid,status FROM mysql.bind_info WHERE original_sql='%s' AND default_db='%s'",
+		originSQL, defaultDB)
+	rs, err := exec.Execute(ctx, sql)
+	if err != nil {
+		return err
+	}
+	if len(rs) == 1 {
+		chkBatch := rs[0].NewRecordBatch()
+		for {
+			err = rs[0].Next(context.TODO(), chkBatch)
+			if err != nil {
+				return err
+			}
+			if chkBatch.NumRows() == 0 {
+				break
+			}
+			it := chunk.NewIterator4Chunk(chkBatch.Chunk)
+			for row := it.Begin(); row != it.End(); row = it.Next() {
+				rowID := row.GetInt64(0)
+				status := row.GetString(1)
+				if status == using {
```

Comment:
We should drop the old one before creating a new one?

Do we decide to support the syntax like `bind if not exists`?

Reply:
No, the deleted status helps DBA manage the bind table, check which bind has been deleted and the deleted time

Reply:
No, this status is our DBA's suggest. It can help DBA to check history bind info, and the time created or deleted